bitwarden_license/src/Scim/Queries/Groups/GetGroupQuery.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Scim.Models;
+using Bit.Scim.Queries.Groups.Interfaces;
+
+namespace Bit.Scim.Queries.Groups;
+
+public class GetGroupQuery : IGetGroupQuery
+{
+    private readonly IGroupRepository _groupRepository;
+
+    public GetGroupQuery(IGroupRepository groupRepository)
+    {
+        _groupRepository = groupRepository;
+    }
+
+    public async Task<ScimGroupResponseModel> GetGroupAsync(Guid organizationId, Guid groupId)
+    {
+        var group = await _groupRepository.GetByIdAsync(groupId);
+        if (group == null || group.OrganizationId != organizationId)
+        {
+            throw new NotFoundException("Group not found.");
+        }
+
+        return new ScimGroupResponseModel(group);
+    }
```

Comment:
I'm somewhat hesitant that we get any value of of this change. We're leaking controller concerns here.

We could perhaps write it as:

```suggestion
    public async Task<ScimGroupResponseModel> GetGroupAsync(Guid organizationId, Guid groupId)
    {
        var group = await _groupRepository.GetByIdAsync(groupId);
        if (group == null || group.OrganizationId != organizationId)
        {
            return null;
        }

        return group;
    }
```

But at that point I wonder if we're better of just keeping it in the controller.

Reply:
I like that approach, that way the command could be reused elsewhere without being tied to the controller response type. IMO it still makes sense keeping the command - this way we can separate the business logic from the API.
@eliykat thoughts?

Reply:
Moving the `ScimGroupResponseModel` up to the controller level is a good suggestion. And I can understand why we may not throw here, it seems like more reasonable default behaviour to return null and then the caller can throw if it wants.

But then we also have to add a null check & throw to the controller, at which point we're just doubling up on code. I agree with Oscar that we may as well just move it back into the controller. We'll have to do the same for Users/GET as well. (I just eyeballed the rest of the controller methods, I still think they're all fine to extract out - these GET methods are the simplest.)

Reply:
That said, I doubt that many of these SCIM queries/commands are going to be generally applicable/reusable. Many of them take SCIM request models and/or have SCIM-specific logic (see #2268 as an example of both) and will therefore be tied to SCIM. Unless you want to keep much of the logic in the controller and only use queries/commands for more generalizable operations (e.g. restore/revoke/delete user, which are mostly handled by services at the moment anyway).

I basically expected these refactors to be a lift & shift of existing code out of the controller, with maybe a few minor adjustments. If it's more than that then we might need your review @Hinton or (better yet) to pair with myself and/or @r-tome to go through them.

Reply:
I've been thinking about this further - I'd say the argument for keeping it in the controller is just how little code it is. I wouldn't expect most of these SCIM queries/commands to be reusable in other contexts, because most will take SCIM models or be doing SCIM-specific operations (e.g. #2268), which is fine.

Reply:
I've updated the branch with your suggestion!